src/HttpClientPoolItem.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Http\Client\Common;
+
+use Http\Client\HttpAsyncClient;
+use Http\Client\HttpClient;
+use Psr\Http\Message\RequestInterface;
+use Http\Client\Exception;
+
+/**
+ * A HttpClientPoolItem represent a HttpClient inside a Pool.
+ *
+ * It is disabled when a request failed and can be reenable after a certain number of seconds
+ * It also keep tracks of the current number of request the client is currently sending (only usable for async method)
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+class HttpClientPoolItem implements HttpClient, HttpAsyncClient
```

Comment:
> i could see the use for customizing the pool item behaviour. disable is protected, indicating this is meant to be extended for custom use cases.

Yes, i don't want to add that here, but i can see use case where you want to persist metadata (disable, counter of request) of the HttpClientPoolItem, that's why i want to allow inheritance of this (so we can add, in the futur, an extended library using a storage to share this metadata across php process)